sky/shell/platform/ios/sky_dynamic_service_loader.mm

```diff
@@ -25,6 +28,48 @@ - (SkyDynamicServiceHandler)serviceEntryPoint;
 
 @end
 
+enum class InstallSystemThunksResult {
+  Failure,
+  EmbedderOlder,
+  EmbedderNewer,
+  Success,
+};
+
+static InstallSystemThunksResult InstallSystemThunksInLibrary(
+    void* library_handle) {
+  if (library_handle == NULL) {
+    return InstallSystemThunksResult::Failure;
+  }
+
+  dlerror();
```

Comment:
Each call to `dlerror()` returns the last error that happened on the thread and then resets it. Since I am `dlsym`'ing next and checking the `dlerror` again, I like to clear it explicitly in case the implementation of `dlsym` does not clear it in case of no errors. Probably me being too paranoid.